scipy/special/wright_bessel.pxd

```diff
@@ -0,0 +1,685 @@
+# zeta(3)
+DEF M_Z3 = 1.2020569031595942854
+# rgamma_zero: smallest value x for which rgamma(x) == 0 as x gets large
+DEF rgamma_zero = 178.47241115886637
+# exp_inf: smallest value x for which exp(x) == inf
+DEF exp_inf = 709.78271289338403
+# exp_zero: largest value x for which exp(x) == 0
+# DEF exp_zero = -745.13321910194117
+
+
+# 1. Taylor series expansion in x=0, for x <= 1
+#    Phi(a, b, x) = sum_k x^k / k! / Gamma(a*k+b)
+#
+# Note that every term, and therefore also Phi(a, b, x), is monotone decreasing
+# with increasing a or b.
+@cython.cdivision(True)
+cdef inline double _wb_series(double a, double b, double x,
+    unsigned int nstart, unsigned int nstop) nogil:
+    cdef:
+        unsigned int k, k_max
+        double xk_k, res
+
+    xk_k = x**nstart * Gamma(nstart + 1)  # x^k/k!
+    res = xk_k * rgamma(nstart*a + b)
+    # term k=nstart+1 , +2 +3, ...
+    if nstop > nstart:
+        # series expansion until term k such that a*k+b <= rgamma_zero
+        k_max = int(floor((rgamma_zero - b) / a))
+        if nstop > k_max:
+            nstop = k_max
+        for k in range(nstart + 1, nstop):
+            xk_k *= x/k
+            res += xk_k * rgamma(a*k + b)
+
+    return res
+
+
+# 2. Taylor series expansion in x=0, for large a.
+#    Phi(a, b, x) = sum_k x^k / k! / Gamma(a*k+b)
+#
+#    Use Stirling formula to find k=k_max, the maximum term.
+#    Then use n terms of Taylor series around k_max.
+@cython.cdivision(True)
+cdef inline double _wb_large_a(double a, double b, double x,
+    unsigned int n) nogil:
+    cdef:
+        unsigned int k, k_max
+        int n_start
+        double lnx, res
+
+    k_max = int(pow(pow(a, -a) * x, 1./(1 + a)))
+    n_start = k_max - n//2
+    if n_start < 0:
+        n_start = 0
+
+    res = 0
+    lnx = log(x)
+    for k in range(n_start, n_start + n):
+        res += exp(k * lnx - lgam(k + 1) - lgam(a*k + b))
+
+    return res
+
+
+# 3. Taylor series in a=0 up to order 5, for tiny a and not too large x
+# Phi(a, b, x) = exp(x)/Gamma(b)
+#               * (1 - a*x*Psi(b) + a^2/2*x*(1+x)(Psi(b)^2 - Psi'(b)) + O(a^3))
+# where Psi is the digamma function.
+# Call: python _precompute/wright_bessel.py 1
+#
+# For small b, cancellation of poles of digamma(b)/Gamma(b) and polygamma needs
+# to be carried out => series expansion in a and b around 0 to order 5.
+# Call: python _precompute/wright_bessel.py 2
+@cython.cdivision(True)
+cdef inline double _wb_small_a(double a, double b, double x, int order) nogil:
+    cdef:
+        double dg, dg1, dg2, dg3, res
+        double A[6]  # powers of a^k/k!
+        double B[5]  # powers of b^k/k!
+        double C[5]  # coefficients of a^k1 * b^k2
+        int k
+
+    if b <= 1e-3:
+        # Series expansion of both a and b up to order 5:
+        C[0] = 1
+        C[1] = 2*M_EG
+        C[2] = 3*M_EG**2 - M_PI**2/2
+        C[3] = 4*M_EG**3 - 2*M_EG*M_PI**2 + 8*M_Z3
+        C[4] = 5*M_EG**4 - 5*M_EG**2*M_PI**2 + 40*M_EG*M_Z3 + M_PI**4/12
```

Comment:
Better to write out the decimals here (precomputed with mpmath) to ensure they are correct to the last bit. (`gcc` will do precompute the things involving `M_PI` to arbitrary precision for you, but not for `M_EG`.)